cpp/src_prims/stats/weighted_mean.cuh

```diff
@@ -56,6 +58,49 @@ void rowWeightedMean(
     [WS] __device__(Type v) { return v / WS; });
 }
 
+/**
+ * @brief Compute the row-wise weighted mean of the input matrix
```

Comment:
This could use a better description, and the one above could be expanded just a tad to denote the differences between these two functions. 
```suggestion
 * @brief Compute the row-wise weighted mean of the input matrix with a vector with a vector of sample weights. 
```